factory_linux.go

```diff
@@ -192,6 +192,18 @@ func (l *LinuxFactory) Type() string {
 	return "libcontainer"
 }
 
+func (l *LinuxFactory) Destroy() error {
+	ls, err := ioutil.ReadDir(l.Root)
+	if err != nil {
+		return newGenericError(err, SystemError)
+	}
+	if len(ls) != 0 {
+		return newGenericError(fmt.Errorf("%s is not empty", l.Root), RootIsNotEmpty)
+	}
+	syscall.Unmount(l.Root, syscall.MNT_DETACH)
```

Comment:
Ya, but why is unmount even here at all?  shouldn't it be configurable and only called if you passed the option?  Like we need to register destroy funcs that are called at destroy time?